nixos/modules/services/security/vault.nix

```diff
@@ -1,207 +1,132 @@
-          type =  types.str;
-          default = "inMemory";
-          description = ''
-            The name of the type of storage backend.
-          '';
-        };
-
-        extraConfig = mkOption {
-          type = types.lines;
-          default = "";
-          description = ''
-            Configuration for storage
-          '';
-        };
- 
+      tlsCertFile = mkOption {
+        type = types.str;
+        default = "/etc/vault/cert.pem";
+        example = "/path/to/your/cert.pem";
+        description = "TLS certificate file. A self-signed certificate will be generated if file not exists";
       };
 
+      tlsKeyFile = mkOption {
+        type = types.str;
+        default = "/etc/vault/key.pem";
+        example = "/path/to/your/key.pem";
+        description = "TLS private key file. A self-signed certificate will be generated if file not exists";
+      };
 
-      telemetry = {
-
-        disable_hostname = mkOption {
-          type =  types.bool;
-          default = false;
-          description = ''
-            Specifies if gauge values should be prefixed with the local hostname. 
-          '';
-        };
+      listenerExtraConfig = mkOption {
+        type = types.lines;
+        default = ''
+          tls_min_version = "tls12"
+        '';
+        description = "extra configuration";
+      };
 
-        extraConfig = mkOption {
-          type = types.lines;
-          default = "";
-          description = ''
-            configuration for telemetry
-          '';
-        };        
+      storageBackend = mkOption {
+        type = types.enum ["inmem" "consul" "zookeeper" "file" "s3" "azure" "dynamodb" "etcd" "mssql" "mysql" "postgresql" "swift" "gcs"];
+        default = "inmem";
+        description = "The name of the type of storage backend";
+      };
 
+      storageConfig = mkOption {
+        type = types.lines;
+        description = "Storage configuration";
+        default = "";
       };
 
+      telemetryConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "Telemetry configuration";
+      };
     };
-
   };
 
   config = mkIf cfg.enable {
 
-    systemd.services.vault =
-    { description = "Vault server daemon";
-
-      wantedBy = ["multi-user.target"];
+    users.extraUsers.vault = {
+      name = "vault";
+      group = "vault";
+      uid = config.ids.uids.vault;
+      description = "Vault daemon user";
+    };
+    users.extraGroups.vault.gid = config.ids.gids.vault;
 
-      preStart =
-        ''
-          mkdir -m 0755 -p /var/lib/vault
-        '';
+    systemd.services.vault = {
+      description = "Vault server daemon";
 
-      serviceConfig =
-        { ExecStart =
-            "${pkgs.vault}/bin/vault server -config ${configFile}";
-          KillMode = "process";
-        };
-    };   
+      wantedBy = ["multi-user.target"];
+      after = [ "network.target" ];
+
+      preStart = ''
+        mkdir -m 0755 -p /var/lib/vault
```

Comment:
I am not sure if this directory is used somehow, it might be a rudiment of the original PR.